src/orderbook.ts

```diff
@@ -0,0 +1,34 @@
+export class Orderbook {
+  from: string;
+  to: string;
+  asks: Map<number, number>;
+  bids: Map<number, number>;
+
+  constructor(from: string, to: string) {
+    this.from = from;
+    this.to = to;
+    this.asks = new Map();
+    this.bids = new Map();
+  }
+
+  addBid(price: number, volume: number) {
+    addOffer(price, volume, this.bids);
+  }
+
+  addAsk(price: number, volume: number) {
+    addOffer(price, volume, this.asks);
+  }
+
+  toJSON(): string {
+    const asks = Array.from(this.asks.entries());
+    const bids = Array.from(this.bids.entries());
+    asks.sort((lhs, rhs) => lhs[0] - rhs[0]);
+    bids.sort((lhs, rhs) => lhs[0] - rhs[0]);
+    return JSON.stringify({bids, asks});
```

Comment:
It looks the `toJSON` does 2 things:
- get the book in a representable way
- stringify it

So I would split it in two functions. Actually, I would only implement the first one, since `JSON.stringify` can be done by the user of this order book.

Reply:
Also, I'm not a big fan of acronyms like lhs and rhs. 

what about:
```js
Array.from(this.bids.entries())
   .sort(([leftPrice], [rightPrice]) => leftPrice - rightPrice)
```

Also, maybe u can create a function so you can reuse it for both sortings.

Reply:
👍

Reply:
Turns out toJSON should not serialize, just return the struct you want later to be serialized (https://github.com/douglascrockford/JSON-js/blob/master/json2.js). Will change it so that we return the struct and leave it up to the caller to serialize.